block.go

```diff
@@ -0,0 +1,52 @@
+package sdk
+
+import (
+	"context"
+
+	ctypes "github.com/cometbft/cometbft/rpc/core/types"
+	cosmos "github.com/cosmos/cosmos-sdk/client"
+)
+
+// NodeStatusFetcher returns the status of the node to which it is connected.
+//
+// This status is used here in the BlockClient in order to get the latest block height.
+type NodeStatusFetcher interface {
+	Status(ctx context.Context) (*ctypes.ResultStatus, error)
+}
+
+// BlockClient is the interface to interact with the on-chain block module.
+//
+// For example, it can be used to get the latest block height.
+//
+// It uses the StatusClient interface of the cometbft to pull the latest block height.
+// This is done to avoid fetching the entire latest block just to extract the block height.
+//
+// The implementations of this interface could leverage websockets to get notified
+// about new blocks and cache the latest block height to avoid querying the blockchain
+// for it every time.
+type BlockClient struct {
+	NodeStatusFetcher
+}
+
+// NewBlockClient creates a new block client with the provided RPC URL.
+func NewBlockClient(queryNodeRPCUrl string) (*BlockClient, error) {
```

Comment:
I'm pretty confused by the whole BlockClient/HeightClient/NodeStatusFetcher split and organization. I see no pattern.

Reply:
Since we are decoupling the implementation from the **consumer** interface, the implementation (`BlockClient`) being the API producer uses the `NodeStatusFetcher` to achieve its functionality.

While the API consumer (`SDK`) is defining its needs through an API consumer interface which effectively needs a `HeightClient`.

In a nutshell:
* The consumer (`SDK`) describing its needs: "I need a client that provides me a height" .
* The producer (`BlockClient`) describing its features: "I know how to provide blocks and part of my functionality is compatible with your needs (Providing height)"